src/App.tsx

```diff
@@ -1,14 +1,59 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { SortBy } from './types/SortBy';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [sortBy, setSortBy] = useState<SortBy>(SortBy.All);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [hasLoadingError, setHasLoadingError] = useState(false);
+
+  const loadTodos = useCallback(async () => {
+    try {
+      setTodos(await getTodos());
+      setHasLoadingError(false);
+    } catch (error) {
+      if (error instanceof Error) {
```

Comment:
better to have state that handle error add in case of error we could show user some message etc
this catch crash your App it's not good for UX